internal/services/appconfiguration/app_configuration_resource.go

```diff
@@ -325,6 +357,28 @@ func resourceAppConfigurationCreate(d *pluginsdk.ResourceData, meta interface{})
 	}
 	meta.(*clients.Client).AppConfiguration.AddToCache(resourceId, *resp.Model.Properties.Endpoint)
 
+	expandedReplicas, err := expandAppConfigurationReplicas(d.Get("replica").(*schema.Set).List(), name, location)
+	if err != nil {
+		return fmt.Errorf("expanding `replica`: %+v", err)
+	}
+
+	replicaClient := meta.(*clients.Client).AppConfiguration.ReplicasClient
+	for _, replica := range expandedReplicas {
+		replicaId := replicas.NewReplicaID(subscriptionId, resourceGroup, name, *replica.Name)
```

Comment:
To be consistent we should use the values from the parsed id 
```suggestion
		replicaId := replicas.NewReplicaID(id.SubscriptionId, id.ResourceGroupName, id.Name, *replica.Name)
```

Reply:
fixed, thanks.